src/Npgsql/Internal/ResolverFactories/AdoTypeInfoResolverFactory.cs

```diff
@@ -76,9 +76,9 @@ static TypeInfoMappingCollection AddMappings(TypeInfoMappingCollection mappings)
             // Text
             // Update PgSerializerOptions.IsWellKnownTextType(Type) after any changes to this list.
             mappings.AddType<string>(DataTypeNames.Text,
-                static (options, mapping, _) => mapping.CreateInfo(options, new StringTextConverter(options.TextEncoding), preferredFormat: DataFormat.Text), isDefault: true);
+                static (options, mapping, _) => mapping.CreateInfo(options, new StringTextConverter(options.TextEncoding, nested: false), preferredFormat: DataFormat.Text), isDefault: true);
```

Comment:
Nestedness is not a mapping invariant. For instance this converter will be composed over by the array converter (via AddArrayType) and it would do the wrong thing. You'd have to look at the writer (we could maybe have some state there) to know whether the converter is called in a top-level context.

Reply:
Ok I'll take this PR back to draft for a while. I'll have to get a better understanding of the mapping to know what the next steps are.